qutip/visualization.py

```diff
@@ -328,51 +381,58 @@ def hinton(rho, x_basis=None, y_basis=None, color_style="scaled",
 
     fig, ax = _is_fig_and_ax(fig, ax)
 
-    # Extract plotting data W from the input.
-    if isinstance(rho, Qobj):
-        if rho.isoper or rho.isoperket or rho.isoperbra:
-            if rho.isoperket:
-                rho = vector_to_operator(rho)
-            elif rho.isoperbra:
-                rho = vector_to_operator(rho.dag())
-            W = rho.full()
-            # Create default labels if none are given.
-            if x_basis is None or y_basis is None:
+    if not isinstance(rho, list):
+        rhos = [rho]
+    else:
+        rhos = rho
+
+    Ws = list()
+    w_max = 0
+    for rho in rhos:
```

Comment:
Should we ask for the rho's dims to match?
Otherwise it will not make a continuous animation.

Reply:
I added an inner function to assert shapes.